src/schema/v2/quiz.ts

```diff
@@ -34,6 +35,55 @@ export const QuizType = new GraphQLObjectType<any, ResolverContext>({
         }
       },
     },
+    recommendedArtworks: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ArtworkType))
+      ),
+      resolve: async (
+        { quiz_artworks },
+        _,
+        { savedArtworkLoader, relatedLayerArtworksLoader }
+      ) => {
+        if (!savedArtworkLoader) return []
+
+        try {
+          const artworks = await Promise.all(
+            quiz_artworks.map(({ artwork_id }) => {
+              return savedArtworkLoader(artwork_id)
+            })
+          )
+
+          const savedArtworks = artworks.filter(({ is_saved }) => is_saved)
```

Comment:
or rather, here there can be one loder/API call to return which of the quiz artworks are saved, vs. the individual fetch+filter

Reply:
Believe it or not, this is correct.